provider/aws/heartbeat.go

```diff
@@ -12,9 +15,28 @@ func (p *Provider) Heartbeat() (map[string]interface{}, error) {
 		return nil, err
 	}
 
+	ns, err := p.Cluster.CoreV1().Nodes().List(context.TODO(), am.ListOptions{})
+	if err != nil {
+		return nil, errors.WithStack(err)
+	}
+
+	spotCnt := 0
```

Comment:
hey @nightfury1204, I think to avoid logic duplication, you can put this code just in > providers/k8s/k8s.go. Or create a different file and call inside `heartbeat()` function.

Reply:
provider wise it's different so it will be here. I will put this into different function.